### PR TITLE
DT-2847: ✨ Warn the user if they have no privileges in the requested …

### PR DESCRIPTION
**Pull request by petergphillips**
…service

### CODE REVIEW

File: src/main/resources/db/dev/data/auth/V900_0__oauth_sample_data.sql [hunk 1]
```diff
@@ -115,7 +115,7 @@ VALUES ('BOOK_MOVE', 'Book a secure move', 'Book a secure move', 'ROLE_PECS_SUPP
        ('POM', 'Allocate a POM Service', 'Allocate the appropriate offender manager to a prisoner', 'ROLE_ALLOC_MGR', 'https://moic.service.justice.gov.uk', 1, 'https://moic.service.justice.gov.uk/help'),
        ('PATHFINDER', 'Pathfinder Service', 'View and Manage Pathfinder nominals', 'ROLE_PF_STD_PRISON,ROLE_PF_APPROVAL,ROLE_PF_POLICE', 'http://localhost:3000', 1, null),
        ('SOC', 'Manage SOC cases', 'View and manage SOC cases', 'ROLE_SOC_CUSTODY,ROLE_SOC_COMMUNITY', 'http://localhost:3000', 1, null),
-       ('USERADMIN', 'Manage user accounts', null, 'ROLE_KW_MIGRATION,ROLE_MAINTAIN_ACCESS_ROLES,ROLE_MAINTAIN_ACCESS_ROLES_ADMIN,ROLE_MAINTAIN_OAUTH_USERS,ROLE_AUTH_GROUP_MANAGER', 'http://localhost:3001/', 1, null),
+       ('manage-user-accounts-ui', 'Manage user accounts', null, 'ROLE_KW_MIGRATION,ROLE_MAINTAIN_ACCESS_ROLES,ROLE_MAINTAIN_ACCESS_ROLES_ADMIN,ROLE_MAINTAIN_OAUTH_USERS,ROLE_AUTH_GROUP_MANAGER', 'http://localhost:3001/', 1, null),
```
Comment (petergphillips) [inline src/main/resources/db/dev/data/auth/V900_0__oauth_sample_data.sql:118]: changed so can map between client and service

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/UserContextService.kt [hunk 2]
```diff
@@ -32,19 +32,35 @@ class UserContextService(
...
-  private fun findEnabledUsersWithRoles(email: String, source: AuthSource, roles: List<String>) = findUsers(email, source)
-    .filter { u -> u.isEnabled }
-    .filter { roles.isEmpty() || it.authorities.map { r -> r.authority }.intersect(roles.toSet()).isNotEmpty() }
+  fun checkUser(
+    loginUser: UserPersonDetails,
+    scopes: Set<String> = emptySet(),
+    roles: List<String> = emptyList(),
+  ): Boolean {
+    val requestedSources = findRequestedSources(scopes)
+    return requestedSources.contains(AuthSource.fromNullableString(loginUser.authSource)) &&
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/UserContextService.kt:49]: Not sure if I understand why source is relevant to this process

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/UserContextService.kt (hunk 2 above)
Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/UserContextService.kt:49]: so we added functionality for Delius to auth for azure logins whereby we are only interested in allowing access to the service if they have the delius authSource.  This is defined in the client in the scopes field, so if the scope is set to one of delius , nomis or auth then the user must have a matching authSource.

**Pull request merged**